src/picongpu/include/simulationControl/MySimulation.hpp

```diff
@@ -462,7 +462,7 @@ class MySimulation : public SimulationHelper<simDim>
         auto fieldE = dc.get< FieldE >( FieldE::getName(), true );
         auto fieldB = dc.get< FieldB >( FieldB::getName(), true );
 
-        if( step != 0 )
+        if( this->restartRequested )
         {
             namespace nvfct = PMacc::nvidia::functors;
 
```

Comment:
@psychocoderHPC Please see the tests in #2137 - especially the one without the fix - it shows an equivalent behavior as with this commit. 

I will try to reproduce your background field now, to test restarts.